pywhy_graphs/algorithms/tests/test_cg.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+
+from pywhy_graphs import CG
+from pywhy_graphs.algorithms import is_valid_cg
+
+
+@pytest.fixture
+def cg_simple_partially_directed_cycle():
+    graph = CG()
+    graph.add_nodes_from(["A", "B", "C", "D"])
+    graph.add_edge("A", "B", graph.directed_edge_name)
+    graph.add_edge("D", "C", graph.directed_edge_name)
+    graph.add_edge("B", "D", graph.undirected_edge_name)
+    graph.add_edge("A", "C", graph.undirected_edge_name)
+
+    return graph
+
+
+@pytest.fixture
+def cg_multiple_blocks_partially_directed_cycle():
+
+    graph = CG()
+    graph.add_nodes_from(["A", "B", "C", "D", "E", "F", "G"])
+    graph.add_edge("A", "B", graph.directed_edge_name)
+    graph.add_edge("D", "C", graph.directed_edge_name)
+    graph.add_edge("B", "D", graph.undirected_edge_name)
+    graph.add_edge("A", "C", graph.undirected_edge_name)
+    graph.add_edge("E", "F", graph.undirected_edge_name)
+    graph.add_edge("F", "G", graph.undirected_edge_name)
+    graph.add_edge("G", "E", graph.undirected_edge_name)
+
+    return graph
+
+
+@pytest.fixture
+def square_graph():
+    graph = CG()
+    graph.add_nodes_from(["A", "B", "C", "D"])
+    graph.add_edge("A", "B", graph.undirected_edge_name)
+    graph.add_edge("B", "C", graph.undirected_edge_name)
+    graph.add_edge("C", "D", graph.undirected_edge_name)
+    graph.add_edge("C", "A", graph.undirected_edge_name)
+
+    return graph
+
+
+@pytest.fixture
+def fig_g1_frydenberg():
```

Comment:
Perhaps add a docstring and a link to the reference for all the fixtures that come from some paper, so its back traceable?

Reply:
yes that's a good idea, i have the reference in the function but not the test